scylla/src/frame/value.rs

```diff
@@ -252,6 +271,20 @@ impl Value for Uuid {
     }
 }
 
+impl Value for BigInt {
+    fn serialize(&self, buf: &mut Vec<u8>) -> Result<(), ValueTooBig> {
+        let bytes_num_pos: usize = buf.len();
+
+        buf.put_i32(0);
```

Comment:
I don't think that the trick with writing zero at the beginning and overwriting it with correct size at the end is necessary here. It's easy to calculate the size up front:

```rust
let serialized = self.to_signed_bytes_be();
let serialized_len: i32 = serialized.len().try_into().map_err(|_| ValueTooBig)?;

buf.put_i32(serialized_len);
buf.extend_from_slice(&serialized);

Ok(())
```

Similar can be done for `BigDecimal`, too.

Reply:
Done :)